site/docs/plugins/commands.md

```diff
@@ -5,10 +5,628 @@ next: false
+The commands plugin helps you deal with that by allowing you to suggest a command that might be what the user wanted in the first place.
+It is compatible with custom prefixes, so you don't have to worry about that, and its usage is quite straightforward:
+
+::: code-group
+
+```ts [TypeScript]
+// Use the flavor to create a custom context
+type MyContext = Context & CommandsFlavor;
+
+// Use the new context to instantiate your bot
+const bot = new Bot<MyContext>("token");
+const myCommands = new CommandGroup<MyContext>();
+
+// ... Register the commands
+
+bot
+  // Check if there is a command
+  .filter(commandNotFound(myCommands))
+  // If so, that means it wasn't handled by any of our commands.
+  .use(async (ctx) => {
+    // We found a potential match
+    if (ctx.commandSuggestion) {
+      await ctx.reply(
+        `Hmm... I don't know that command. Did you mean ${ctx.commandSuggestion}?`,
+      );
+    }
+
+    // Nothing seems to come close to what the user typed
+    await ctx.reply("Oops... I don't know that command :/");
+  });
+```
+
+```js [JavaScript]
+// Use the new context to instantiate your bot
+const bot = new Bot("token");
+const myCommands = new CommandGroup();
+
+// ... Register the commands
+
+bot
+  // Check if there is a command
+  .filter(commandNotFound(myCommands))
+  // If so, that means it wasn't handled by any of our commands.
+  .use(async (ctx) => {
+    // We found a potential match
+    if (ctx.commandSuggestion) {
+      await ctx.reply(
+        `Hmm... I don't know that command. Did you mean ${ctx.commandSuggestion}?`,
+      );
+    }
+
+    // Nothing seems to come close to what the user typed
+    await ctx.reply("Oops... I don't know that command :/");
+  });
+```
+
+:::
+
+Behind the scenes, `commandNotFound` will use the `getNearestCommand` context method which by default will prioritize commands that correspond to the user language.
+If you want to opt-out of this behavior, you can pass the `ignoreLocalization` flag set to true.
+It is possible to search across multiple CommandGroup instances, and `ctx.commandSuggestion` will be the most similar command, if any, across them all.
+It also allows to set the `ignoreCase` flag, which will ignore casing while looking for a similar command and the `similarityThreshold` flag, which controls how similar a command name has to be to the user input for it to be recommended.
+
+The `commandNotFound` function will only trigger for updates which contain command-like text similar to your registered commands.
+For example, if you only have registered [commands with a custom prefix](#prefix) like `?`, it will trigger the handler for anything that looks like your commands, e.g: `?sayhi` but not `/definitely_a_command`.
+Same goes the other way, if you only have commands with the default prefix, it will only trigger on updates that look like `/regular` `/commands`.
```

Comment:
```suggestion
Same goes the other way, if you only have commands with the default prefix, it will only trigger on updates that look like `/regular` and `/commands`.
```